src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseGradingService.java

```diff
@@ -473,7 +460,7 @@ private void updateScore(Result result, Set<ProgrammingExerciseTestCase> success
             }
 
             // The score is calculated as a percentage of the maximum points
-            long score = Math.round(successfulTestPoints / maxScoreRespectingZeroPointExercises * 100.0);
+            long score = Math.round(successfulTestPoints / programmingExercise.getMaxScore() * 100.0);
```

Comment:
here we should use the round() function of utils.ts and provide an additional round function which send back the percentage score

Reply:
Agree, only that this should be some kind of RoundUtil.java file ;-)